core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java

```diff
@@ -94,6 +94,7 @@
   private final Map<Class<? extends PluginTypeInterface>, Set<PluginTypeListener>> listeners = new HashMap<>();
 
   private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+  private int waitForPluginToBeAvailableLimit = 2000;
```

Comment:
Should be a static final constant, with upper case / underscore naming convention.